src/packcc.c

```diff
@@ -3886,6 +3886,11 @@ static bool_t generate(context_t *ctx) {
             "    for (i = 0; i < count; i++) pcc_lr_table_entry__destroy(auxil, table->buf[i]);\n"
             "    memmove(table->buf, table->buf + count, sizeof(pcc_lr_table_entry_t *) * (table->len - count));\n"
             "    table->len -= count;\n"
+            "    for (i = 0; i < table->len; i++) {\n"
+            "        if (table->buf[i]) {\n"
+            "            table->buf[i]->memos.buf->answer->pos -= count;\n"
```

Comment:
I'm not very familiar with the internals of these structures. It's possible that I didn't include some check or something here. Please review this carefully before merging.

Reply:
It should be correct.
I was impressed with your quick work!